### PR TITLE
Hide scrollbars while resizing

### LINKED ISSUE #23: UX: Hide scrollbars while resizing
When minimizing a split layout, it will show the native scrollbar (Window & Linux)

Found the issue in UX testing #13

### PR DESCRIPTION
**Pull request by limonte**
Fixes #23 

Inspired by: http://demo.vaadin.com/sampler/#ui/structure/horizontal-split-panel

Before:
recorded

After:
recorded

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-split-layout/21)

### CODE REVIEW

**Comment by samiheikki** [comment]
Was just about to write a ticket about this issue 😄

**Comment by Saulis** [comment]
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[vaadin-split-layout.html, line 125 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/21#-KXKU-2RYNxMIcK_cUlv:-KXKU-2RYNxMIcK_cUlw:b-joztm2) ([raw file](https://github.com/vaadin/vaadin-split-layout/blob/05183f83a4ccdc30c7b5f84d881cc82dcdff0ec7/vaadin-split-layout.html#L125)):*
> ```HTML
>     </style>
>     <content select="[vaadin-split-layout=primary]" id="primary"></content>
>     <div id="splitter" on-track="_onHandleTrack" on-down="_onDown" on-up="_onUp">
> ```

Instead of listening to `down` and `up` events, would this work if we instead handle all this inside the `_onHandleTrack(e)` ?
Basically, the current overflow styles would be stored in the beginning of `_onHandleTrack` and then we have a debouncer inside which will restore the style after tracking is finished. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/21)*

**Comment by limonte** [comment]
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[vaadin-split-layout.html, line 125 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/21#-KXKU-2RYNxMIcK_cUlv:-KXL1W8K9yCD8iNzIhNg:b-896fix) ([raw file](https://github.com/vaadin/vaadin-split-layout/blob/05183f83a4ccdc30c7b5f84d881cc82dcdff0ec7/vaadin-split-layout.html#L125)):*
<details><summary><i>Previously, Saulis (Sauli Tähkäpää) wrote…</i></summary>
> Instead of listening to `down` and `up` events, would this work if we instead handle all this inside the `_onHandleTrack(e)` ?
> Basically, the current overflow styles would be stored in the beginning of `_onHandleTrack` and then we have a debouncer inside which will restore the style after tracking is finished. 
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/21)*

**Comment by tomivirkki** [comment]
Tried out with an alternative suggestion where the issue is fixed by adding:
```
:host {
  overflow: hidden !important;
}
```
to the component styles.

The rule is marked important because like in the "Layout Combination" demo, a split-layout can be a child of another split-layout, which in turn tries to set its overflow to 'auto' (IMO it's too intrusive and could be removed).

Also added default flex for children to avoid jumpyness. Try here: https://github.com/vaadin/vaadin-split-layout/tree/fix/emerging-scrollbars

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/21#-:-KXLtm_u_Hx0qOh45E8S:b-f5hjqj)*

**Comment by platosha** [comment]
One part of this issue is that if you drag the inner splitter very close to the edge, you can trigger a scrollbar in the outer split layout container. As I’ve realised today, this could be easily fixed by setting `overflow: hidden;` on the split-layout hosting elements.

Also, there is a scope problem with the content containers here. The split layout doesn’t introduce internal content containers in its local DOM, but rather uses the external user-provided light DOM elements directly. In the styles, there is some light DOM styling involved, that assigns `overflow: auto;` for the user-provided containers:

```css
:host ::content > * {
  /* ... */
  overflow: auto;
}
```

And this is already pretty intrusive to assign the overflow CSS style for the user-provided elements.

But triggering the `overflow` dynamically with JS in the inline styles of the user-provided elements, this feels too much. Users are basically forced to loose control over the `overflow` property then.

---

Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/21#-:-KXLj9j6fFABCUq-llfP:bjesy5b)*

**Comment by limonte** [comment]
> But triggering the overflow dynamically with JS in the inline styles of the user-provided elements, this feels too much. Users are basically forced to loose control over the overflow property then.

After some thinking I should agree with you.

---

Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/21#-:-KXPSjpj7Fjz3q4rWgvM:bg346tr)*

**Comment by tomivirkki** [comment]
Actually, `vaadin-split-layout` itself is a good example of a component that gets messed up by the intrusive `overlay:auto` rule of `vaadin-split-layout` (this is the cause for the original issue) :)

It would probably make most sense to wrap the `<content>` tags to wrapper elements and target the `overlay:auto` rule to the wrappers, leaving the light DOM children as they are.

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/21#-:-KXPSYcwgrWY5X_p9D1P:b-ha7rh0)*

**Comment by limonte** [comment]
This is fixed in cleaner way in https://github.com/vaadin/vaadin-split-layout/pull/25

**Comment by tomivirkki** [comment]
#25 still has the problem. If you don't want to re-open this PR then create an issue about it so it doesn't get forgotten.

**Comment by limonte** [comment]
> #25 still has the problem. If you don't want to re-open this PR then create an issue about it so it doesn't get forgotten.

Done: https://github.com/vaadin/vaadin-split-layout/issues/26
